trieste/acquisition/function.py

```diff
@@ -544,161 +548,21 @@ def sample(self, at: TensorType) -> TensorType:
         produce the exact same samples. Calls to :meth:`sample` on *different*
         :class:`IndependentReparametrizationSampler` instances will produce different samples.
 
-        :param at: Where to sample the predictive distribution, with shape `[..., D]`, for points
+        :param at: Where to sample the predictive distribution, with shape `[..., 1, D]`, for points
             of dimension `D`.
-        :return: The samples, of shape `[..., S, L]`, where `S` is the `sample_size` and `L` is the
-            number of latent model dimensions.
-        :raise ValueError (or InvalidArgumentError): If ``at`` is a scalar.
+        :return: The samples, of shape `[..., S, 1, L]`, where `S` is the `sample_size` and `L` is
+            the number of latent model dimensions.
+        :raise ValueError (or InvalidArgumentError): If ``at`` has an invalid shape.
         """
-        tf.debugging.assert_rank_at_least(at, 1)
-        mean, var = self._model.predict(at[..., None, :])  # [..., 1, L], [..., 1, L]
+        tf.debugging.assert_shapes([(at, [..., 1, None])])
+        mean, var = self._model.predict(at[..., None, :, :])  # [..., 1, 1, L], [..., 1, 1, L]
```

Comment:
shapes seem weird here

Reply:
it's to allow broadcasting with eps. Is there a cleaner way?

Reply:
we end up with 3 dims anyway

Reply:
or do you think i've got the shapes wrong?

Reply:
It might be that with the batch dimension we do not need to introduce an extra dimension

Reply:
forget it, I think I was wrong